src/exe/colmap.cc

```diff
@@ -417,15 +417,17 @@ int RunDelaunayMesher(int argc, char** argv) {
 }
 
 int RunPatchMatchStereo(int argc, char** argv) {
-#ifndef CUDA_ENABLED
-  std::cerr << "ERROR: Dense stereo reconstruction requires CUDA, which is not "
-               "available on your system."
+#if !defined(CUDA_ENABLED) && !defined(TORCH_ENABLED)
```

Comment:
Logic here is changed such that now we fail immediately only if both CUDA and Torch are missing. If either is present we can do patch-match through the existing method or PMNet